### PR TITLE
E2E tests

### PR DESCRIPTION
**Pull request by tmashuang**
Add E2E tests with TokenFactory, adding token, sending eth, and add a Firefox build. Also, changes the path for getting the chrome extension id in newer version of chrome.

### CODE REVIEW

File: test/screens/new-ui.js [hunk 1]
```diff
@@ -39,7 +39,7 @@ async function captureAllScreens() {
   const extPath = path.resolve('dist/chrome')
   driver = buildWebDriver(extPath)
   await driver.get('chrome://extensions-frame')
-  const elems = await driver.findElements(By.css('.extension-list-item-wrapper'))
+  const elems = await driver.findElements(By.css('* /deep/ extensions-item'))
```
Comment (kumavis) [inline test/screens/new-ui.js]: `/deep/` combinator is deprecated

**Review by kumavis — CHANGES_REQUESTED** [review]

File: test/screens/new-ui.js (hunk 1 above)
Comment (tmashuang) [reply test/screens/new-ui.js]: Darn. I will find an alternative, but coming up empty on solutions to getting the local extension id.

**Comment by danfinlay** [comment]
We don't really want our tests hitting the network. Ideally we would have a local build of a dapp (like TokenFactory) in our test suite, and test it against a TestRPC instance, so even offline, we could verify MetaMask worked well.

**Comment by kumavis** [comment]
@danfinlay this hits against ganache-cli

File: test/screens/new-ui.js (hunk 1 above)
Comment (kumavis) [reply test/screens/new-ui.js]: @tmashuang its likely different on chrome and ff, eh?

Comment (kumavis) [reply test/screens/new-ui.js]: oh i see this is for chrome only atm

Comment (kumavis) [reply test/screens/new-ui.js]: there may be a better way to do this but
```
document.querySelector('extensions-manager').shadowRoot.querySelector('extensions-view-manager extensions-item-list').shadowRoot.querySelector('extensions-item')
```

Comment (kumavis) [reply test/screens/new-ui.js]: here is a single selector, annoying tho
```js
document.querySelector('extensions-manager::shadow extensions-view-manager extensions-item-list::shadow extensions-item')
```

Comment (tmashuang) [reply test/screens/new-ui.js]: but `::shadow` is also deprecated, also `>>>`

Comment (kumavis) [reply test/screens/new-ui.js]: damn

Comment (kumavis) [reply test/screens/new-ui.js]: so then the multistep one above with `.shadowRoot` I guess

File: test/e2e/chrome/metamask.spec.js [hunk 2]
```diff
@@ -107,7 +102,7 @@ describe('Metamask popup page', function () {
 
     it('shows value was created and seed phrase', async () => {
       await delay(300)
-      seedPhase = await driver.findElement(By.css('.twelve-word-phrase')).getText()
+      await driver.findElement(By.css('.twelve-word-phrase')).getText()
```
Comment (kumavis) [inline test/e2e/chrome/metamask.spec.js]: dont need to do this if we're not using the value

Comment (kumavis) [reply test/e2e/chrome/metamask.spec.js]: maybe get the value and verify the content is 12 words
`seedPhrase.split(' ').length === 12`

**Comment by kumavis** [comment]
#4229 also includes a Fix for `test/screens/new-ui.js`
https://github.com/MetaMask/metamask-extension/pull/4229/files#diff-1b94881ca708ecfcb9b85a4174bbf731

**Comment by kumavis** [comment]
we should be able to dedupe most of `test/e2e/chrome/metamask.spec.js`
and `test/e2e/firefox/metamask.spec.js`

**Comment by kumavis** [comment]
pull `develop` for deps fix

**Review by kumavis — APPROVED** [review]

**Comment by Alexander1984z** [comment]
Спасибо за поддержку и работу

**Pull request merged**